openbas-api/src/test/java/io/openbas/rest/AtomicTestingApiTest.java

```diff
@@ -121,40 +104,92 @@ void duplicateAndDeleteAtomicTestingTest() throws Exception {
+  @WithMockAdminUser
+  void launchAtomicTesting() throws Exception {
+    AtomicTestingInput atomicTestingInput =
+        AtomicTestingInputFixture.createDefaultAtomicTestingInput();
+    atomicTestingInput.setInjectorContract(INJECTOR_CONTRACT.getId());
+
+    String createdInject =
         mvc.perform(
-                delete(ATOMIC_TESTINGS_URI + "/" + NEW_INJECT_ID)
-                    .accept(MediaType.APPLICATION_JSON))
+                post(ATOMIC_TESTINGS_URI)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(asJsonString(atomicTestingInput)))
             .andExpect(status().is2xxSuccessful())
+            .andExpect(jsonPath("$.inject_status.status_name").value("DRAFT"))
             .andReturn()
             .getResponse()
             .getContentAsString();
-    assertNotNull(response);
-    // Assert delete
-    response =
+
+    String injectId = JsonPath.read(createdInject, "$.inject_id");
+
+    mvc.perform(post(ATOMIC_TESTINGS_URI + "/" + injectId + "/launch"))
+        .andExpect(status().is2xxSuccessful())
+        .andExpect(jsonPath("$.inject_status.status_name").value("QUEUING"));
+  }
+
+  @Test
+  @DisplayName("Relaunch an Atomic Testing")
+  @WithMockAdminUser
+  void relaunchAtomicTesting() throws Exception {
+    AtomicTestingInput atomicTestingInput =
+        AtomicTestingInputFixture.createDefaultAtomicTestingInput();
+    atomicTestingInput.setInjectorContract(INJECTOR_CONTRACT.getId());
+
+    String createdInject =
         mvc.perform(
-                get(ATOMIC_TESTINGS_URI + "/" + NEW_INJECT_ID).accept(MediaType.APPLICATION_JSON))
-            .andExpect(status().is4xxClientError())
+                post(ATOMIC_TESTINGS_URI)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(asJsonString(atomicTestingInput)))
+            .andExpect(status().is2xxSuccessful())
+            .andExpect(jsonPath("$.inject_status.status_name").value("DRAFT"))
             .andReturn()
             .getResponse()
             .getContentAsString();
-    assertNotNull(response);
+
+    String injectId = JsonPath.read(createdInject, "$.inject_id");
+
+    mvc.perform(post(ATOMIC_TESTINGS_URI + "/" + injectId + "/launch"))
+        .andExpect(status().is2xxSuccessful())
+        .andExpect(jsonPath("$.inject_status.status_name").value("QUEUING"));
+
+    String relaunchedInject =
+        mvc.perform(post(ATOMIC_TESTINGS_URI + "/" + injectId + "/relaunch"))
+            .andExpect(status().is2xxSuccessful())
+            .andExpect(jsonPath("$.inject_status.status_name").value("QUEUING"))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+    String relaunchedInjectId = JsonPath.read(relaunchedInject, "$.inject_id");
+
+    mvc.perform(get(ATOMIC_TESTINGS_URI + "/" + injectId)).andExpect(status().is4xxClientError());
+
+    mvc.perform(get(ATOMIC_TESTINGS_URI + "/" + relaunchedInjectId))
```

Comment:
In my point of view verifying if the first inject has been deleted by making another HTTP request within the same test undermines the concept of unit testing. It introduces dependencies on the GET request, which could fail independently, making the test less reliable and no longer a true unit test

Reply:
Discussed together, here we are at the Integration testing level. For the relaunch of the Inject, we need to make sure that the original Inject has been deleted. I'm leaving the test as it is.